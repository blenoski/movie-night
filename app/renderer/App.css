body {
  background-color: #141414;
  min-width: 320px;
}

.App {
  background-color: #141414;
  width: 100%;
}

.App-header {
  background-color: rgb(20,20,20);
  background-color: rgba(20,20,20,0.7);
  color: white;
  padding: 10px 10px 10px 20px;
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  font-family: CopperPlate, Times;
  font-size: 1.5em;
  display: flex;
  justify-content: space-between;
  flex-wrap: nowrap;
}

h2 {
  line-height: 1.3;
  margin: 0;
  display: block;
  font-size: 1.2em;
  font-weight: bold;
  color: #999;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  background-color: transparent;
  text-transform: capitalize;
  cursor: pointer;
}

h2:hover {
  color: #FFF;
}

.movie:active {
  border: 5px solid rgba(255,255,255,0.5);
}

.play-button {
  background: rgba(0,0,0,0.1);
  border: 3px solid rgba(255,255,255,0.9);
  border-radius: 50%;
  height: 80px;
  width: 80px;
  padding-left: 5%;
  color: rgba(255,255,255,0.9);
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.play-button:hover {
  color: rgb(2,117,216);
  background-color: rgba(0,0,0,0.5);
}

.mscroll::-webkit-scrollbar {
    height: 10px;
}

.mscroll::-webkit-scrollbar-button {
  display: none;
}

.mscroll::-webkit-scrollbar-track {
    display: none;
}

.mscroll::-webkit-scrollbar-track:hover {
  display: block;
  border-radius: 10px;
  box-shadow: inset 0 0 4px rgba(255,255,255,0.4);
}

.mscroll::-webkit-scrollbar-thumb {
    display: block;
    border-radius: 10px;
    box-shadow: inset 0 0 4px rgba(255,255,255,0.4);
}
