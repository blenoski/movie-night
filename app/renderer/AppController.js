// The AppController is very specific to this application.
// It has knowledge of events, actions, components, etc...
// The job of the AppController is to wire up all the events, actions,
// callbacks, etc...
import { ipcRenderer } from 'electron'
import {
  CRAWL_COMPLETE,
  MOVIE_DATABASE,
  SEARCHING_DIRECTORY,
  SELECT_IMPORT_DIRECTORY
 } from '../shared/events'
import store, { updateMovieDB } from './AppState'
import {
  updateCrawling,
  updateSearchDirectory
} from './import-movies'
import logger from './mainWindowLogger'

// Handle SEARCHING_DIRECTORY events
ipcRenderer.on(SEARCHING_DIRECTORY, function (event, directory) {
  logger.debug('Received SEARCHING_DIRECTORY event', { directory })
  store.dispatch(updateSearchDirectory(directory))
  logger.debug('Dispatched updateSearchDirectory action', { directory })
})

// Handle CRAWL_COMPLETE events
ipcRenderer.on(CRAWL_COMPLETE, function (event, directory) {
  logger.info('Received CRAWL_COMPLETE event', { directory })
  store.dispatch(updateSearchDirectory(directory))
  logger.info('Dispatched updateSearchDirectory action', { directory })
  store.dispatch(updateCrawling(false))
  logger.info('Dispatched updateCrawling(false) action')
})

// Handle MOVIE_DATABASE events
ipcRenderer.on(MOVIE_DATABASE, (event, movieDB) => {
  logger.info('Recieved MOVIE_DATABASE event', { count: movieDB.length })
  store.dispatch(updateMovieDB(movieDB))
  logger.info('Dispatched updateMovieDB action', { count: movieDB.length })
})

// Invoking this function will kick off the import movies workflow.
function importMovies () {
  ipcRenderer.send(SELECT_IMPORT_DIRECTORY)
  logger.info('Sent SELECT_IMPORT_DIRECTORY event')
}

// API
const AppController = {
  importMovies
}

export default AppController
