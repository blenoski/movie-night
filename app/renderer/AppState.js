import { createStore, applyMiddleware, combineReducers } from 'redux'
import {
  reducer as displayMoviesReducer,
  stateKey as displayMoviesStateKey
} from './display-movies'
import {
  reducer as importMoviesReducer,
  stateKey as importMovieStateKey
} from './import-movies'
import {
  reducer as searchMoviesReducer,
  stateKey as searchMovieStateKey
} from './search-movies'

// Set-up the combined redux reducers.
let reducers = {}
reducers[displayMoviesStateKey] = displayMoviesReducer
reducers[importMovieStateKey] = importMoviesReducer
reducers[searchMovieStateKey] = searchMoviesReducer
const allReducers = combineReducers(reducers)

// Create the redux store
const createStoreWithMiddleware = applyMiddleware()(createStore)
const store = createStoreWithMiddleware(allReducers)

export default store
