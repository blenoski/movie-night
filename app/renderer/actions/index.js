import { SEARCH_DIRECTORY } from './types'

export function updateSearchDirectory (directory) {
  return {
    type: SEARCH_DIRECTORY,
    payload: directory
  }
}
