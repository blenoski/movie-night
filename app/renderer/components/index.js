import cn from 'classnames'
import React from 'react'
import 'bootstrap/dist/css/bootstrap.css'

export const Button = ({className, outline, primary, ...props}) => {
  return <button
    type='button'
    className={cn(
      'btn',
      primary && !outline && 'btn-primary',
      primary && outline && 'btn-outline-primary',
      className
    )}
    {...props}
  />
}

export const TextInput = ({className, ...props}) => {
  return <input
    type='text'
    className={cn(
      'form-control input',
      className
    )}
    {...props}
  />
}
