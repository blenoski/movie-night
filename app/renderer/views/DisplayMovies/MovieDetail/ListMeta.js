import React from 'react'
import styled from 'styled-components'

export default ({ actors, director, genres }) =>
  <Section>
    <Item>
      <Heading>Starring:</Heading> {actors.join(', ')}
    </Item>
    <Item>
      <Heading>Director:</Heading> {director}
    </Item>
    <Item>
      <Heading>Genres:</Heading> {genres.join(', ')}
    </Item>
  </Section>

const Section = styled.section`
  margin-top: 24px;
`

const Heading = styled.span`
  color: rgba(255, 255, 255, 0.9);
`

const Item = styled.h6`
  text-transform: capitalize;
`
