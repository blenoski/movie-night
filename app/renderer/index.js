import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import App from './App'
import store from './AppState'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
