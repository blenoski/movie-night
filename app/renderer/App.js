import React, { Component } from 'react'
import styled from 'styled-components'
import AppController from './AppController'
import DisplayMovies from './display-movies'
import ImportMovies from './import-movies'
import Logo from './logo'
import SearchMovies from './search-movies'

export default class App extends Component {
  render () {
    return (
      <Application>
        <Header>
          <Logo />
          <AppControls>
            <SearchMovies />
            <ImportMovies onClick={AppController.importMovies} />
          </AppControls>
        </Header>
        <DisplayMovies />
      </Application>
    )
  }
}

const Application = styled.div`
  background: rgb(20,20,20);
  padding-top: 40px;
`

const Header = styled.header`
  background-color: rgba(20,20,20,0.7);
  color: white;
  display: flex;
  flex-wrap: nowrap;
  font-family: CopperPlate, Times;
  font-size: 1.5em;
  justify-content: space-between;
  left: 0;
  right: 0;
  top: 0;
  padding: 10px 10px 10px 20px;
  position: fixed;
`

const AppControls = styled.div`
  display: flex;
`
