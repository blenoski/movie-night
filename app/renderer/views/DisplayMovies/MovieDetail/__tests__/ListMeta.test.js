'use strict'
/* globals describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import ListMeta from '../ListMeta'

describe('ListMeta', () => {
  test('renders metadata', () => {
    const component = renderer.create(
      <ListMeta
        actors={['Ben Stiller', 'Tom Hanks', 'Jessica Alba']}
        director={'Todd Phillips'}
        genres={['action', 'comedy', 'drama']}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    // snapshot captures display of actors, director, and genres
  })
})
