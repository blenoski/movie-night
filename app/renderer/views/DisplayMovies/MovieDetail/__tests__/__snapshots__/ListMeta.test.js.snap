// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMeta renders metadata 1`] = `
<section
  className="sc-bdVaJa fyrQKn"
>
  <h6
    className="sc-htpNat jeDJyY"
  >
    <span
      className="sc-bwzfXH fyPPwG"
    >
      Starring:
    </span>
     
    Ben Stiller, Tom Hanks, Jessica Alba
  </h6>
  <h6
    className="sc-htpNat jeDJyY"
  >
    <span
      className="sc-bwzfXH fyPPwG"
    >
      Director:
    </span>
     
    Todd Phillips
  </h6>
  <h6
    className="sc-htpNat jeDJyY"
  >
    <span
      className="sc-bwzfXH fyPPwG"
    >
      Genres:
    </span>
     
    action, comedy, drama
  </h6>
</section>
`;
